### PR TITLE
Front end login and auth context [WIP]

### PR DESCRIPTION
**Pull request by MorganLove**
**DESCRIPTION**

- Adds authentication context for the front end
- Adds login buttons for the AA and JA flows

**New Tests**

- none yet

**Checklist**

*Function*

[] Code is fully functional and ready for review (or is tagged WIP)

[] No errors/warnings in the browser console ( ﾟヮﾟ)/

[] package.json is committed if I added or removed any dependencies

[] Tests added as needed to ensure code base still has 100% passing coverage

[] Tests added as needed to cover new functionality

*Style*

[] Passes linter without errors

[] Code is easily understandable and self-documenting (using naming conventions in keeping with the rest of the codebase)

[] Interfaces all have 'I' before their names (as in `IAudit`)

[] Comments are only included as needed to explain the 'why' instead of the 'what'

[] Common standards and coding practices are adhered to

[] DRY: Don't Repeat Yourself. Code is reasonably structured to avoid repetition

[] Code is intuitively structured following a Separation of Concerns

[] No class components; only functional components (using Hooks as needed)

*Workflow*

[] Do an interactive rebase or a rebase

### CODE REVIEW

**Comment by MorganLove** [comment]
@benadida It looks to me like it should work, but when I click the login buttons it redirects back to the home page and the backend logs the requests as 302s. (I am using `window.location.replace` instead of `Link` or an `a` element or `history.push()` because all of those were just returning a 404, for some reason). Any ideas where the problem might be?

File: arlo-client/src/components/CreateAudit.tsx [hunk 1]
```diff
@@ -44,17 +47,48 @@ const CreateAudit = ({ history }: RouteComponentProps<ICreateAuditParams>) => {
...
+          <Button
+            type="button"
+            intent="primary"
+            fill
+            large
+            onClick={onClick}
+            loading={loading}
+            disabled={loading}
+          >
+            Create a New Audit
```
Comment (benadida) [inline arlo-client/src/components/CreateAudit.tsx]: as per a discussion with @mcchilders, let's let a non-authenticated user create a new audit, too. So the create audit should be functional in both cases, not-logged-in and logged-in.

Comment (MorganLove) [reply arlo-client/src/components/CreateAudit.tsx]: @benadida Gotcha! Can do.

Comment (MorganLove) [reply arlo-client/src/components/CreateAudit.tsx]: @benadida Should audits created by a non-authenticated user be publicly accessible, like the old flow?

**Pull request merged**